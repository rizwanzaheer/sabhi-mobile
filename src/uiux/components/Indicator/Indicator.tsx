import * as React from 'react'
import { ViewStyle } from 'react-native'
import Container from '../Container/Container'
import Text, { TextTypes } from '../Text/Text'
import { withTheme } from '../../theming'

interface IndicatorBarProps {
  /**
   * Indicator text
   */
  text: string

  /**
   * Indicator background
   */
  backgroundColor?: string

  /**
   * Theming
   */
  theme: any
}

const IndicatorBar: React.FC<IndicatorBarProps> = ({ text, backgroundColor, theme }) => {
  const triangle: ViewStyle = {
    width: 0,
    height: 0,
    backgroundColor: 'transparent',
    borderStyle: 'solid',
    borderLeftWidth: 20,
    borderRightWidth: 20,
    borderBottomWidth: 15,
    borderLeftColor: 'transparent',
    borderRightColor: 'transparent',
    borderBottomColor: backgroundColor ? backgroundColor : '#222222',
    transform: [{ rotate: '180deg' }],
  }

  return (
    <Container backgroundColor={theme.colors.primary.background}>
      <Container backgroundColor={backgroundColor ? backgroundColor : '#222222'} padding={true}>
        <Text textColor={'#FFFFFF'} type={TextTypes.SectionHeader} textAlign={'center'}>
          {text}
        </Text>
      </Container>
      <Container alignItems={'center'}>
        <Container viewStyle={triangle} />
      </Container>
    </Container>
  )
}

export default withTheme(IndicatorBar)
