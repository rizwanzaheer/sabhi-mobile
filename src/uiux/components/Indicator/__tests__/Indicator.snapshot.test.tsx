import * as React from 'react'
import { render } from 'react-native-testing-library'
import Indicator from '../Indicator'

describe('Component(snapshot): Indicator', () => {
  it('renders with default props', () => {
    const tree = render(<Indicator text={'Test indicator text'} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with background color', () => {
    const tree = render(<Indicator text={'Test indicator text'} backgroundColor={'#000000'} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
