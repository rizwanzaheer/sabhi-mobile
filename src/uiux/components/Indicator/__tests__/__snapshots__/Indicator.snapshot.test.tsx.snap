// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshot): Indicator renders with background color 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#FFFFFF",
      "borderRadius": undefined,
      "flex": undefined,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": "#000000",
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": 15,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "flexWrap": "wrap",
          "fontSize": 14,
          "textAlign": "center",
        }
      }
    >
      Test indicator text
    </Text>
  </View>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": "transparent",
          "borderBottomColor": "#000000",
          "borderBottomWidth": 15,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 20,
          "borderRadius": undefined,
          "borderRightColor": "transparent",
          "borderRightWidth": 20,
          "borderStyle": "solid",
          "flex": undefined,
          "flexDirection": undefined,
          "height": 0,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "transform": Array [
            Object {
              "rotate": "180deg",
            },
          ],
          "width": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Component(snapshot): Indicator renders with default props 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#FFFFFF",
      "borderRadius": undefined,
      "flex": undefined,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": "#222222",
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": 15,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "flexWrap": "wrap",
          "fontSize": 14,
          "textAlign": "center",
        }
      }
    >
      Test indicator text
    </Text>
  </View>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": "transparent",
          "borderBottomColor": "#222222",
          "borderBottomWidth": 15,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 20,
          "borderRadius": undefined,
          "borderRightColor": "transparent",
          "borderRightWidth": 20,
          "borderStyle": "solid",
          "flex": undefined,
          "flexDirection": undefined,
          "height": 0,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "transform": Array [
            Object {
              "rotate": "180deg",
            },
          ],
          "width": 0,
        }
      }
    />
  </View>
</View>
`;
